utils/bimap/bimap.go

```diff
@@ -100,3 +114,28 @@ func (m *BiMap[K, V]) DeleteValue(val V) (K, bool) {
 func (m *BiMap[K, V]) Len() int {
 	return len(m.keyToValue)
 }
+
+func (m *BiMap[K, V]) MarshalJSON() ([]byte, error) {
+	return json.Marshal(m.keyToValue)
+}
+
+func (m *BiMap[K, V]) UnmarshalJSON(b []byte) error {
```

Comment:
Where is the custom unmarshaller used?

Reply:
or is this just moreso ensuring if we add marshal, we should also have unmarshal?

Reply:
no where. I can remove it if you'd prefer not to have it. Although I felt like implementing json marshal on a normal data structure and not supporting unmarshaling was a bit weird.